test/libmpv_test.c

```diff
@@ -15,15 +15,256 @@
+                                              enum mpv_format formats[])
+{
+    for (int i = 0; options[i]; i++) {
+        switch (formats[i]) {
+        case MPV_FORMAT_STRING:
+            check_api_error(mpv_set_option(ctx, options[i], formats[i], &str));
+            check_api_error(mpv_set_property(ctx, properties[i], formats[i], &str));
+            break;
+        case MPV_FORMAT_FLAG:
+            check_api_error(mpv_set_option(ctx, options[i], formats[i], &flag));
+            check_api_error(mpv_set_property(ctx, properties[i], formats[i], &flag));
+            break;
+        case MPV_FORMAT_INT64:
+            check_api_error(mpv_set_option(ctx, options[i], formats[i], &int_));
+            check_api_error(mpv_set_property(ctx, properties[i], formats[i], &int_));
+            break;
+        case MPV_FORMAT_DOUBLE:
+            check_api_error(mpv_set_option(ctx, options[i], formats[i], &double_));
+            check_api_error(mpv_set_property(ctx, properties[i], formats[i], &double_));
+            break;
+        }
+    }
+}
+
+static void test_file_loading(char *file)
+{
+    const char *cmd[] = {"loadfile", file, NULL};
+    check_api_error(mpv_command(ctx, cmd));
+    int loaded = 0;
+    int finished = 0;
+    time_t end = time(NULL) + 10;
+    while (time(NULL) < end) {
+        if (finished)
+            break;
+        mpv_event *event = mpv_wait_event(ctx, 0);
+        switch (event->event_id) {
+        case MPV_EVENT_FILE_LOADED:
+            // make sure it loads before exiting
+            loaded = 1;
+            break;
+        case MPV_EVENT_END_FILE:
+            if (loaded)
+                finished = 1;
+            break;
+        }
+    }
+    if (!finished)
+        fail("Unable to load test file!\n");
+}
+
+static void test_lavfi_complex(char *file)
+{
+    const char *cmd[] = {"loadfile", file, NULL};
+    check_api_error(mpv_command(ctx, cmd));
+    int finished = 0;
+    int loaded = 0;
+    time_t end = time(NULL) + 10;
+    while (time(NULL) < end) {
+        if (finished)
+            break;
+        mpv_event *event = mpv_wait_event(ctx, 0);
+        switch (event->event_id) {
+        case MPV_EVENT_FILE_LOADED:
+            // Add file as external and toggle lavfi-complex on.
+            if (!loaded) {
+                check_api_error(mpv_set_property_string(ctx, "external-files", file));
+                const char *add_cmd[] = {"video-add", file, "auto", NULL};
+                check_api_error(mpv_command(ctx, add_cmd));
+                check_api_error(mpv_set_property_string(ctx, "lavfi-complex", "[vid1] [vid2] vstack [vo]"));
+            }
+            loaded = 1;
+            break;
+        case MPV_EVENT_END_FILE:
+            if (loaded)
+                finished = 1;
+            break;
+        }
+    }
+    if (!finished)
+        fail("Lavfi complex failed!\n");
+}
+
+static void test_options_and_properties(void)
+{
+    // Order matters. string -> flag -> int -> double (repeat)
```

Comment:
I'm missing a tiny comment where that describes what we're actually testing

Reply:
Put it a couple lines above this (github seems to cut it off from the view).